modules/travel_pay/app/models/travel_pay/base_expense.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module TravelPay
+  class BaseExpense
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+    include ActiveModel::Validations
+    include ActiveModel::Serialization
+
+    attribute :purchase_date, :datetime
+    attribute :description, :string
+    attribute :cost_requested, :float
+    attribute :claim_id, :string
+
+    validates :purchase_date, presence: true
+    validates :description, presence: true, length: { maximum: 255 }
+    validates :cost_requested, presence: true, numericality: { greater_than: 0 }
+
+    # Custom belongs_to association with Claim
+    #
+    # @return [Object, nil] the associated claim object or nil if not found
+    def claim
+      return nil unless claim_id
+
+      @claim ||= find_claim_by_id(claim_id)
+    end
+
+    # Setter for claim association
+    # Accepts a claim object and extracts its ID
+    #
+    # @param claim_obj [Object] the claim object to associate
+    def claim=(claim_obj)
+      @claim = claim_obj
+      self.claim_id = claim_obj&.id
+    end
+
+    # Custom has_one association with Receipt
+    #
+    # @return [Object, nil] the associated receipt or nil
+    def receipt_association
+      receipt
+    end
+
+    # Receipt attribute accessor
+    attr_accessor :receipt
+
+    # Returns a hash representation of the expense
+    # Useful for JSON serialization and API responses
+    #
+    # @return [Hash] hash representation of the expense
+    def as_json(_options = {})
+      result = attributes.dup
+      result['claim_id'] = claim_id
+      result['has_receipt'] = receipt.present?
+      result['receipt'] = receipt
+      result['expense_type'] = expense_type
+      result
```

Comment:
The result is not json, it's a hash - can we call this `to_h` to align with Ruby convention, and handle the json serialization outside of the model?

Reply:
Yeah, no problem. Do you still want these additional properties being appended (like `has_receipt`) or is that something you feel is more appropriately handled outside of the model as well?